ggu/main.c

```diff
@@ -356,7 +356,7 @@ submit_ggu(struct rte_mbuf **pkts, unsigned int num_pkts,
 	}
 
 	req->num_pkts = num_pkts;
-	rte_memcpy(req->pkts, pkts, sizeof(*req->pkts) * num_pkts);
+	req->pkts = pkts;
```

Comment:
BUG: This assignment is invalid. It assumes that the caller of `submit_ggu()` has allocated heap memory for `pkts`. But if you follow the chain of calls, `pkts` is allocated in the execution stack and will go away before the GGU unit have a chance to process the message. By the way, the caller is `process_acl()`.

Reply:
The same issue applies to the CPS patch.

Reply:
And to the LLS patch.

Reply:
Thanks @AltraMayor. You are right. The ``pkts`` are defined in ``process_acl()`` as ``struct rte_mbuf *pkts[astate->func_count][acl->num];``, which goes away once the call is done.